classes/local/store/s3/client.php

```diff
@@ -390,6 +395,26 @@ public function support_presigned_urls() {
      * @return string.
      */
     public function generate_presigned_url($contenthash, $headers) {
+        $client = 'cf' === $this->signingmethod ? 'cloudfront' : 's3';
+        return  $this->{"generate_presigned_url_$client"}($contenthash, $headers);
```

Comment:
I generally dislike code like this because it makes it harder to find stuff. If you search for generate_presigned_url_s3 then you won't find where it is called from